### ISSUE TITLE
Issue with arrays/objects

### ISSUE THREAD

**Issue by dts** [ericelliott/riteway#190]
Currently, the deep equality isn't verifying types, so arrays and objects are treated as "equal" even though they are not.  Here's an example:

```
const { Try, describe } = require('riteway');

describe('array-object-fail',async (test) => {
  test({
    given: 'an array and an object',
    should: 'not be equal',
    expected: {},
    actual: []
  });

  test({
    given: 'another example',
    should: 'fail',
    expected: { 0: 'foo' },
    actual: ['foo'],
  });
});
```

**Comment by cbbfcd** [comment]
i will fixed it, see my pr.

**Comment by ericelliott** [comment]
Fixed in riteway@6.1.2
